play-akka-http-ws-standalone/src/main/scala/play/api/libs/ws/akkahttp/StandaloneAkkaHttpWSResponse.scala

```diff
@@ -0,0 +1,82 @@
+package play.api.libs.ws.akkahttp
+
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import play.api.libs.ws.{ StandaloneWSResponse, WSCookie }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+private[akkahttp] object StandaloneAkkaHttpWSResponse {
+  def apply(resp: HttpResponse)(implicit mat: Materializer) = new StandaloneAkkaHttpWSResponse(resp)
+}
+
+final class StandaloneAkkaHttpWSResponse private (val response: HttpResponse)(implicit val mat: Materializer) extends StandaloneWSResponse {
+
+  final val UnmarshalTimeout = 1.second
+
+  /**
+   * Return the current headers for this response.
+   */
+  override def headers: Map[String, Seq[String]] =
+    response.headers
+      .groupBy(_.name)
+      .mapValues(_.map(_.value()))
+
+  /**
+   * Get the underlying response object.
+   */
+  override def underlying[T]: T = ???
+
+  /**
+   * The response status code.
+   */
+  override def status: Int = response.status.intValue()
+
+  /**
+   * The response status message.
+   */
+  override def statusText: String = ???
+
+  /**
+   * Get all the cookies.
+   */
+  override def cookies: Seq[WSCookie] = ???
+
+  /**
+   * Get only one cookie, using the cookie name.
+   */
+  override def cookie(name: String): Option[WSCookie] = ???
+
+  /**
+   * The response body decoded as String, using a simple algorithm to guess the encoding.
+   *
+   * This decodes the body to a string representation based on the following algorithm:
+   *
+   *  1. Look for a "charset" parameter on the Content-Type. If it exists, set `charset` to its value and go to step 3.
+   *  2. If the Content-Type is of type "text", set charset to "ISO-8859-1"; else set `charset` to "UTF-8".
+   *  3. Decode the raw bytes of the body using `charset`.
+   *
+   * Note that this does not take into account any special cases for specific content types. For example, for
+   * application/json, we do not support encoding autodetection and will trust the charset parameter if provided..
+   *
+   * @return the response body parsed as a String using the above algorithm.
+   */
+  override def body: String = {
+    import akka.http.scaladsl.unmarshalling.PredefinedFromEntityUnmarshallers.stringUnmarshaller
+    Await.result(Unmarshal(response).to[String], UnmarshalTimeout)
```

Comment:
I would rather suggest that we do a `toStrict` on the response, and store the strictified entity.
The current impl would blow up if we do `r.body; r.body` I think?

Reply:
Yes, I agree, that will only work once. We first need to clarify the semantics whether these methods can be called several times and then figure out how to be able to make it work.

Reply:
I think play people value convenience a lot.in this API... perhaps we can do some cheating with a CAS ing state machine here? like, here spin a future and await hehe other to drain? a hack, but it would get us to ship this WS impl. and once we do the smarter things in Akka http we could remove the hack here?

Reply:
It could probably just be a lazy val that collects all the stuff. The question is in which way, so that the below methods would also work. That's what I meant with "clarify the semantics", i.e. whether it is allowed to call one or more of those body interpreting/accessing methods and whether it is allowed to call them only once or several times.

Reply:
Yea, current impl will blow up with such a testcase and needs to be fixed.

I am not sure about calling toStrict every time. If the user is using the client for streamed responses and is materializing stream from `bodyAsSource` only once then we should not waste resources with by calling `toStrict` internally. Take a look at ... where I store strict response only when `getBody` or `getBodyBytes` is called.